pam/internal/adapter/pwquality_c.go

```diff
@@ -0,0 +1,44 @@
+package adapter
+
+// #cgo pkg-config: pwquality
+// #include <stdlib.h>
+// #include <pwquality.h>
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+)
+
+// checkChallengeQuality checks the quality of the new password using the pwquality library.
+func checkChallengeQuality(old, new string) error {
+	pwq := C.pwquality_default_settings()
+	if pwq == nil {
+		return errors.New("could not allocate pw quality default settings")
+	}
+	defer C.pwquality_free_settings(pwq)
+
+	var auxErr *C.char
+	auxErrPointer := unsafe.Pointer(auxErr)
+
+	// Load pwquality configuration (from /etc/security/pwquality.conf)
+	if ret := C.pwquality_read_config(pwq, nil, &auxErrPointer); ret < 0 {
+		var buf [C.PWQ_MAX_ERROR_MESSAGE_LEN]C.char
+		errMsg := C.GoString(C.pwquality_strerror(&buf[0], C.size_t(len(buf)), ret, auxErrPointer))
+		return fmt.Errorf("can't ready pwquality configuration: %s", errMsg)
```

Comment:
```suggestion
		return errors.New(errMsg)
```

Reply:
IIRC, in this case their provided message is not clear enough so we had to add more context.

Reply:
Yeah I agree they're pretty generic: https://github.com/libpwquality/libpwquality/blob/80deb29da767961676fedd87dbf264acc438a720/src/error.c#L146-L149

So better to go this way.